cdap-api-common/src/main/java/io/cdap/cdap/api/data/format/LRUCache.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.api.data.format;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.PriorityQueue;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * A basic implementation of LRU cache. It uses a map to store cache values and a priority queue for tracking
+ * last used time. The implementation only use pure Java class. Due to the limitation of the Java
+ * priority queue of not having support for updating priority without removing/adding element, the implementation
+ * is O(n) to the number of entries in the cache, hence it is more small cache (n < 100).
+ *
+ * This class is thread safe.
+ *
+ * @param <K> type of the key
+ * @param <V> type of the value
+ */
+@ThreadSafe
+final class LRUCache<K, V> {
+
+  private final Map<K, V> entries;
+  private final PriorityQueue<TimedKey<K>> orderedKey;
```

Comment:
nit: plural (orderedKeys) is easier to read/understand

Reply:
Fixed